### PR TITLE
add yt video association algorithm

### PR DESCRIPTION
**Pull request by Exortions**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Maths/FindMaxRecursion.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/**
```
Comment (raklaptudirm) [inline Maths/FindMaxRecursion.js:1]: Remove these changes from this pull request.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: Search/test/YoutubeVideoAssociation.test.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  const algorithm = new AssociationAlgorithm(10)
+
+  test('The video should not be associated after one search', () => {
+    algorithm.search(search)
+
+    expect(algorithm.isVideoAssociated('dQw4w9WgXcQ', 'Rick Roll')).toBe(false)
+  })
+
+  test('The video should be associated after 11 searches', () => {
```
Comment (appgurueu) [inline Search/test/YoutubeVideoAssociation.test.js:22]: You don't even test for that though?

**Review by appgurueu — CHANGES_REQUESTED** [review]
I'm very wary of this "algorithm". It's pretty much a counter dict applied to YT vids, which seems rather odd to me. Why would this be specific to YT or videos in general?

**Comment by Exortions** [comment]
> I'm very wary of this "algorithm". It's pretty much a counter dict applied to YT vids, which seems rather odd to me. Why would this be specific to YT or videos in general?

This algorithm uses YouTube as an *example*, but it can be applied to many other broader things. It's just association, but applied in a YouTube context. I could re-write it to make it more broad if you'd like.

**Comment by Exortions** [comment]
I'll fix the changes you requested and place them in a new PR.

**Pull request closed without merging**